### ISSUE TITLE
Disabling CFG and train on latent space.

### ISSUE THREAD

**Issue by Ted-Picovoice** [SWivid/F5-TTS#1149]
### Checks

- [x] This template is only for research question, not usage problems, feature requests or bug reports.
- [x] I have thoroughly reviewed the project documentation and read the related paper(s).
- [x] I have searched for existing issues, including closed ones, no similar questions.
- [x] I am using English to submit this issue to facilitate community communication.

### Question details

Hi,

I'd wish to ask about the necessity of CFG in text-conditioned speech generation.

The experiment that I have was using a pretrained latent space, then use F5-TTS with CFG disabled to train text-to-latent. However, the problem I faced was that the training was extremely bad (e.g. KL metric was extremely high, e.g. 180 where I'm expecting KL divergence to be around 4), essentially the model is ignoring the text condition, and the output audios sounds like unconditional generation (i.e. still the speaker's voice, but just random utterance).

The dataset is a single speaker dataset, and I reduced the F5-TTS model size to be around 5M.

I've tried to debug extensively (e.g. enabling and disabling Sinusoidal PE and/or ConvNeXt in text embedding, and change the MHSA to the one that I know that works), but at this point, I don't think it's a bug, but rather, the flow matching model just can't learn the text condition. In fact, I think for conditional generation, shouldn't we be using cross attention rather than stacking the text condition with the noised vector? This might hides the condition as we used the filler token, so the short text condition sequence is burried in the long audio frame sequence.

Do the authors think it is because that I disabled CFG? I was disabling it for simplicity, and thought CFG is only for quality enhancement.

Thank you!

**Comment by Ted-Picovoice** [comment]
Found a related issue: https://github.com/SWivid/F5-TTS/issues/1122

But problem not yet solved, I'm still in the process of debugging!

**Comment by Ted-Picovoice** [comment]
Problem solved!

I tried a bunch of techniques to overcome this, but unsure which one was the effective one. Essentially we need to be careful with MHSA attention mask, text embedding mask and filler token logic, model size compared to target dimension. This should get you something at least merely intelligible. As for the rest, i.e. naturalness, we just need to finetune model hyperparameters and architectures...

To conclude, the problem was F5-TTS can't find alignment between text and speech, so it ignores the text and do unconditional generation, but it can be overcomed by techniques with text condition. We don't need CFG for at least intelligible conditional generation.
